src/WebJobs.Extensions.DurableTask/LocalGrpcListener.cs

```diff
@@ -154,27 +159,53 @@ public override Task<Empty> Hello(Empty request, ServerCallContext context)
+                    };
 
-                    // TODO: Ideally, we'd have a single method in the taskhubClient that can handle both scheduled and non-scheduled starts.
-                    // TODO: the type of `ScheduledStartTimestamp` is not nullable. Can we change it to `DateTime?` in the proto file?
-                    if (request.ScheduledStartTimestamp != null)
+                    // Create the ExecutionStartedEvent
+                    ExecutionStartedEvent executionStartedEvent = new ExecutionStartedEvent(-1, request.Input)
                     {
-                        instance = await taskhubClient.CreateScheduledOrchestrationInstanceAsync(
-                            name: request.Name, version: request.Version, instanceId: instanceId, input: Raw(request.Input), startAt: request.ScheduledStartTimestamp.ToDateTime());
-                    }
-                    else
+                        Name = request.Name,
+                        Version = request.Version,
+                        OrchestrationInstance = instance,
+                        ScheduledStartTime = request.ScheduledStartTimestamp?.ToDateTime(),
+                    };
+
+                    // Get the parent trace context from CreateInstanceRequest
+                    string? traceParent = request.ParentTraceContext.TraceParent;
+                    string? traceState = request.ParentTraceContext.TraceState;
+
+                    // Create a new activity with the parent context
+                    ActivityContext.TryParse(traceParent, traceState, out ActivityContext parentActivityContext);
+                    Activity? scheduleOrchestrationActivity = StartActivityForNewOrchestration(executionStartedEvent, parentActivityContext);
+
+                    // Schedule the orchestration
+                    try
                     {
-                        instance = await taskhubClient.CreateOrchestrationInstanceAsync(request.Name, request.Version, instanceId, Raw(request.Input));
+                        await this.GetDurabilityProviderToScheduleOrchestration(context).CreateTaskOrchestrationAsync(
+                            new TaskMessage
+                            {
+                                Event = executionStartedEvent,
+                                OrchestrationInstance = instance,
+                            },
+                            this.GetStatusesNotToOverride());
+
+                        // Stop the activity
+                        scheduleOrchestrationActivity?.Stop();
+
+                        return new P.CreateInstanceResponse
+                        {
+                            InstanceId = instance.InstanceId,
+                        };
                     }
-
-                    // TODO: should this not include the ExecutionId and other elements of the taskhubClient response?
-                    return new P.CreateInstanceResponse
+                    catch (InvalidOperationException)
```

Comment:
Isn't this already handled by the `catch` statements in the outer try/catch block? Seems like you should be able to remove this inner try/catch.